Automattic/remote-data-blocks
# Add Tracks Analytics on frontend

### Description

Add Tracks Analytics by using the `@automattic/calypso-analytics` package. With each track event, the following properties will be tracked:

- Properties related to VIP Platform (e.g. org_id, env_id).
- Properties related to Remote Data Blocks (e.g. plugin_version).
- Event specific properties.

**Note:** This tracking will only work if site is hosted on VIP Platform or if the analytics is enabled via filter.

### How to test?

1. Create a site using `vip dev-env` which will have the MU Plugins.
1. Move `remote-data-blocks` plugin to the `plugins` folder.
1. Add filter in `plugin-loader.php` to enable tracking.

    ```php
    add_filter( 'remote_data_blocks_enable_tracks_analytics', '__return_true' );
    ```
1. Checkout [this wp-calypso PR](https://github.com/Automattic/wp-calypso/pull/96007), `build` and `link` the `@automattic/calypso-analytics` package.
1. To allow tracking on local env comment check on [PHP](https://github.com/Automattic/remote-data-blocks/blob/1574d6f729327eda7a10ecfa686c58a3252f8187/inc/Analytics/TracksAnalytics.php#L30C1-L34C4) and [JS](https://github.com/Automattic/remote-data-blocks/blob/1574d6f729327eda7a10ecfa686c58a3252f8187/src/blocks/remote-data-container/utils/tracks.ts#L15C1-L19C3) side.
1. Trigger track events, use "Live View" to verify that the events are getting logged.
